### PR TITLE
Add str.format_map method

### PR DESCRIPTION
**Pull request by doyshinda**
Contributes to #190

### CODE REVIEW

File: vm/src/obj/objstr.rs [hunk 1]
```diff
@@ -1590,6 +1616,36 @@ fn perform_format(
...
+    dict: &PyObjectRef,
+) -> PyResult {
+    let mut final_string = String::new();
+    for part in &format_string.format_parts {
+        let result_string: String = match part {
+            FormatPart::AutoSpec(_) => {
+                return Err(
+                    vm.new_value_error("Format string contains positional fields".to_string())
+                );
+            }
```
Comment (coolreader18) [inline vm/src/obj/objstr.rs]: You can merge this match arm and the previous one by using the `|` character: `FormatPart::AutoSpec(_) | FormatPart::IndexSpec (_, _) => { ... }`

File: tests/snippets/builtin_format_map.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
+from testutils import AssertRaises
+
+
+def test_format_map():
+    # The following tests were performed in Python 3.7.5:
```
Comment (coolreader18) [inline tests/snippets/builtin_format_map.py]: Could you move these tests to `tests/snippets/strings.py`? And you don't need the separate test function; our tests right now are a lot less structured than most Python tests 😁

File: vm/src/obj/objstr.rs (hunk 1 above)
Comment (doyshinda) [reply vm/src/obj/objstr.rs]: :+1:

**Comment by coolreader18** [comment]
Thanks again for contributing!

**Pull request merged**